### DISCUSSION TITLE
Text Classification Project MemoryError

### DISCUSSION THREAD

**Discussion by marcjermaine-pontiveros** [explosion/spaCy#8051] [Help: Coding & Implementations] [answered]
I'm quite new to Natural Language Processing and I have found this nicely designed package for NLP, spaCy v3.0, and I have used this in my project. I've used the [project templates](https://github.com/explosion/projects) for text classification tasks and used a different dataset. However, when used to the entire training and testing set (i.e. my own train.spacy an dev.spacy) it often encounters a MemoryError. Is there any workaround to this? Thanks in advance.

**Comment by polm** [comment, marked as answer]
If you have too much training data to fit in memory, you can use the brand new feature for [streaming training data](https://spacy.io/usage/training#custom-code-readers-batchers). That allows you to use large or even infinite training data. 

If that doesn't work, let us know, and provide more details about your training setup. For example, how many examples do you have, are you training on GPU, what hardware do you have, etc.

**Reply by marcjermaine-pontiveros** [reply]
Thank you so much for the fast response @polm! I'm trying this [feature](https://spacy.io/usage/training#custom-code-readers-batchers) for training. Is there any way to handle large test sets similar to this one? Thanks again!

**Reply by polm** [reply]
The test set isn't handled by the config. If you mean the dev set, used during training, that has to be finite and fit in memory.
